package/boot/uboot-layerscape/files/fsl_ls1012a-frwy-sdboot-uEnv.txt

```diff
@@ -1,8 +1,8 @@
 fdtaddr=0x8f000000
 loadaddr=0x81000000
-fdt_high=0xffffffffffffffff
+fdt_high=0x8fffffff
 initrd_high=0xffffffffffffffff
-sd_boot=mmc read $fdtaddr 7800 800;mmc read $loadaddr 8000 8000;bootm $loadaddr - $fdtaddr
-bootargs=root=/dev/mmcblk0p1 rw rootwait rootfstype=ext4 noinitrd earlycon=uart8250,mmio,0x21c0500 console=ttyS0,115200
+sd_boot=ext4load mmc 0:1 $loadaddr fitImage;bootm $loadaddr
+bootargs=root=/dev/mmcblk0p2 rw rootwait rootfstype=squashfs,ext4 noinitrd earlycon=uart8250,mmio,0x21c0500 console=ttyS0,115200
```

Comment:
To use overlay, the kmod-fs-f2fs and mkf2fs should be enabled in default, right?

Reply:
I use ext4 overlay like other targets. No need kmod-fs-f2fs and mkf2fs.

Reply:
I see... Thanks.
It seemed to be some format issue after squashfs. I will check too.

[    6.912131] mount_root: rootdisk overlay filesystem has not been formatted yet
sh: [    6.920116] mount_root: no usable overlay filesystem found, using tmpfs overlay

Reply:
It should be:

```
[    6.787073] mount_root: rootdisk overlay filesystem has not been formatted yet
[    7.013550] random: mkfs.ext4: uninitialized urandom read (16 bytes read)
[    7.067903] random: fast init done
[   11.519161] EXT4-fs (loop0): mounted filesystem with ordered data mode. Opts: (null)
[   11.526922] ext4 filesystem being mounted at /tmp/overlay supports timestamps until 2038 (0x7fffffff)
[   11.536265] mount_root: overlay filesystem has not been fully initialized yet
[   11.543684] mount_root: switching to ext4 overlay
```
E2fsprogs should be default package:
https://github.com/openwrt/openwrt/blob/0b555f683efa63d9f4b3a9d64f8f028a48d07386/target/linux/layerscape/Makefile#L22
Do You have e2fsprogs added to image?

Reply:
Yes. I double checked. Here is my full log. Not sure what I missed.

May I know which board you were verified on?
I built with below options enabled.
CONFIG_TARGET_MULTI_PROFILE=y
CONFIG_TARGET_PER_DEVICE_ROOTFS=y
CONFIG_TARGET_ALL_PROFILES=y

Thanks.

Reply:
@yangbolu1991 

My board is LS1046ARDB.
I used "make distclean", wiped my whole SD and I reproduced You state.
I reworked last commit and added missing f2fs features. Could You check if everything is ok?

Reply:
Now it's ok. It seems f2fs is used for root partition larger than 100MB, and ext4 is used for root partiton lower than 100MB.

Reply:
@yangbolu1991 

Thank You,